### PR TITLE
task 7 all code, endpoints.json and README.md updated

### PR DESCRIPTION
**Pull request by Fetters83**
All tests, new end point, endpoints.json and README.md files updated.

### CODE REVIEW

File: __tests__/app.test.js [hunk 1]
```diff
@@ -162,4 +162,70 @@ describe('/api/articles/:article_id/comments',()=>{
...
+describe('/api/articles/:article_id/comments',()=>{
+    test('POST:201 when passed an object with a username and body, body text is posted in the comments table to the specified article_id',()=>{
+        const newComment = {username:"rogersop",body:"my first comment"};
+        return request(app)
+        .post('/api/articles/1/comments')
+        .expect(201)
+        .send(newComment)
+        .then(({body})=>{
+            expect(body).toEqual({comment:{body:"my first comment"}})
+            expect(typeof body.comment.body).toBe("string")
```
Comment (roisinoreilly) [inline __tests__/app.test.js:175]: this line isn't necessary, we can already see it's a string

File: __tests__/app.test.js [hunk 2]
```diff
@@ -162,4 +162,70 @@ describe('/api/articles/:article_id/comments',()=>{
...
+        .send(newComment)
+        .then(({body})=>{
+            expect(body.msg).toBe("this article does not exist")
+            
+            
+        })
+    
+    })
+    test('POST:404 when a username of the correct type is passed but does not exist in the user table, an object is returned with a 404 status and error message',()=>{
+        const newComment = {username:"invalidname",body:"my first comment"};
```
Comment (roisinoreilly) [inline __tests__/app.test.js:193]: perfect

File: __tests__/app.test.js [hunk 3]
```diff
@@ -162,4 +162,70 @@ describe('/api/articles/:article_id/comments',()=>{
...
+        .expect(400)
+        .send(newComment)
+        .then(({body})=>{
+            expect(body.msg).toBe("Bad request")
+            
+            
+        })
+    
+    })
+    test('POST:400 when a comment of an incorrect data type is passed, an object is returned with a 404 status and error message',()=>{
```
Comment (roisinoreilly) [inline __tests__/app.test.js:218]: excellent edge case

File: controllers/comments.controllers.js [hunk 4]
```diff
@@ -9,4 +9,14 @@ function getCommentsByArticleId(req,res,next){
     })
 }
 
-module.exports = {getCommentsByArticleId}
\ No newline at end of file
+function postCommentByArticleId(req,res,next){
+    const{article_id} = req.params
+    const commentBody = req.body
+    insertCommentByArticleId(article_id,commentBody).then((comment)=>{
+       res.status(201).send({comment})
```
Comment (roisinoreilly) [inline controllers/comments.controllers.js:16]: lovely

**Review by roisinoreilly — COMMENTED** [review]
excellent work on this ticket and really great error handling. There's just one test missing and thats a 400 for an invalid article id, but other than that everything looks excellent. Keep up the good work!
